[86bztnr9z][carousel] navigatable carusel controls (#1640)
semrush/intergalactic: website/docs/components/carousel/carousel-a11y.md

| Key                         | Function                                                                                                                                                                           |
| --------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `Tab`, `Shift + Tab`        | Moves focus through interactive elements in the carousel.                                                                                                                          |

> | `Enter`                     | Opens slide in modal window if `zoom` property is set to `true`.                                                                                                                   |

Reviewer: hmmm it might be an issue. It has role=button and buttons are supposed to activate on Space

Reviewer 2: Yep, it was some strange behaviour, I've wrote Michael about. If it will finally works without Space scrolling the page, I'll leave the `Space`.

> | `Left Arrow`, `Right Arrow` | Moves focus to the previous or next tab in the slide indicators. If focus is on the last tab, moves focus to the first tab. Shows the slide associated with the newly focused tab. |

Reviewer: ```suggestion
| `Left Arrow`, `Right Arrow` | Activates the previous or next tab in the slide indicators. If the last tab is active, activates the first tab. Shows the slide associated with the activated tab. |
```
I think it's better to avoid the word "focus" here, because as I see the focus is always on the entire indicator row. And maybe ever remove the last sentence, it might be redundant.

Reviewer 2: Yeah! You are definitely right here, thank you ✨

Table: Roles and attributes

| Component / element   | Role & attributes                             | Usage                                                                                                                                                                                |
| --------------------- | --------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
| `Carousel`            | `role="region"`, `roledescription="carousel"` | Defines the carousel and its controls as a landmark region. `roledescription` informs assistive technologies to identify the element as a "carousel".                                |
| `Carousel.Indicators` | `role="tablist"`                              | Indicates that the element serves as a container for a set of tabs.                                                                                                                  |

> |                       | `aria-label="Slides"`                         | Adds a label to a container for a set of tabs.                                                                                                                                       |

Reviewer: @j-mnizhek I've been thinking about this recently, wouldn't the empty cells be an a11y issue because it's not clear that we mean a merged cell here?

Reviewer 2: This is an interesting question! Let me consult with Glasha first, maybe she have had something similar in her experience

> | `Carousel.Indicator`  | `role="tab"`                                  | Indicates the element serves as a tab control. When focused, it is automatically activated, causing its associated tabpanel to be displayed.                                         |

Reviewer: ```suggestion
| `Carousel.Indicator`  | `role="tab"`                                  | Indicates the element serves as a tab control. When active, the associated tabpanel is automatically displayed.                                         |
```
same motivation as above

| `Carousel.Container`  | `aria-live="polite"`                          | Identifies the container element as a live region in the "polite" state, meaning assistive technology users are informed of changes to the region at the next available opportunity. |
| `Carousel.Item`       | `role="tabpanel"`                             | Indicates that the element acts as a tabpanel, controlled by its linked tab.                                                                                                         |
|                       | `aria-roledescription="slide"`                | Indicates the element serves as a tabpanel that is controlled by its associated tab element.                                                                                         |
|                       | `aria-current="active"`                       | Indicates that this element represents the current item within a container or set of related elements.                                                                               |

> | `Carousel.Prev`       | `aria-controls="carouselID"`                  | Refers to the tabpanel element associated with the tab.                                                                                                                              |

Reviewer: ```suggestion
| `Carousel.Prev`       | `aria-controls="carouselID"`                  | Refers to the tabpanel element associated with the previous tab.                                                                                                                              |
```
might be worth clarifying

|                       | `aria-label="Previous slide"`                 | Defines a label for the button that opens previous list item.                                                                                                                        |

> | `Carousel.Next`       | `aria-controls="carouselID"`                  | Refers to the tabpanel element associated with the tab.                                                                                                                              |

Reviewer: ```suggestion
| `Carousel.Next`       | `aria-controls="carouselID"`                  | Refers to the tabpanel element associated with the next tab.                                                                                                                              |
```
same as with Prev
